File: leetcode-submissions/905_Sort_Array_By_Parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
 
     int leftptr=0;
     int rightptr=nums.size()-1;
     while(leftptr<=rightptr){
         if(nums[leftptr]%2!=0 and nums[rightptr]%2==0){
             swap(nums[leftptr],nums[rightptr]);
             leftptr++;
             rightptr--;
         }else{
             leftptr++;
             rightptr--;
         }
     }  
     return nums; 
    }
};

File: leetcode-submissions/905_Sort_Array_By_Parity.ts
function sortArrayByParity(nums: number[]): number[] {
    const ans:number[] = [];
    for(const num of nums){
        if(num%2===0){
        ans.push(num);
        }
    }
    for(const num of nums){
    if(num%2===1){
        ans.push(num);
    }
    }
    return ans ;
};